### ISSUE TITLE
Support passive declarations of queues and exchanges

### ISSUE THREAD

**Issue by OscaTutenchamon** [benjamin-hodgson/asynqp#33]
I'm trying to accomplish:
> Publish to exchange `X` if exists, if not do nothing.

First, I used `passive=True`in `asynqp.spec.ExchangeDeclare`, which produces (as expected) an error in RabbitMQ:
```bash
=ERROR REPORT==== 26-Aug-2015::11:12:47 ===
Channel error on connection <0.1206.0> (127.0.0.1:47403 -> 127.0.0.1:5672, vhost: '/', user: 'guest'), channel 2:
{amqp_error,not_found,"no exchange 'exchange.user.cccc' in vhost '/'",'exchange.declare'}
```
But nothing on python application side:
```python
rmq_chan.sender.send_method(asynqp.spec.ExchangeDeclare(
    0, 
    'exchange.user.cccc',  # name
    'fanout',              # type
    True,                  # passive
    False,                 # durable
    True,                  
    False, 
    # False,               # no_wait  
    {}                     # arguments
))
```
But why bother if we can just publish to non existent exchange with the same effect:
```python
rmq_chan.sender.send_BasicPublish(
    'exchange.user.cccc',
    routing_key='',
    mandatory=False,
    message=asynqp.Message('{"zzz":33}')
)
```
Again, no error in application despite error in RabbitMQ.
```bash
=ERROR REPORT==== 26-Aug-2015::10:53:17 ===
Channel error on connection <0.1206.0> (127.0.0.1:47403 -> 127.0.0.1:5672, vhost: '/', user: 'guest'), channel 1:
{amqp_error,not_found,"no exchange 'exchange.user.cccc' in vhost '/'",'basic.publish'}
```
This would not be a big deal except that `rmq_chan` cannot be used to send any more commands to RabbitMQ after such error (and I don't actually know it happened).

**Comment by benjamin-hodgson** [comment]
`asynqp` is missing a feature - passive declarations of channels and queues - but I think you are also misunderstanding the API.

`asynqp` *does* raise channel errors as Python exceptions, when you use the library correctly. You should be using [`channel.declare_exchange`](http://asynqp.readthedocs.org/en/v0.4/reference.html#asynqp.Channel.declare_exchange) to declare your exchange (and awaiting it using `yield from`). Don't use `channel.sender` - it's an internal and undocumented part of `asynqp`.

Now, `channel.declare_exchange` doesn't presently support a `passive` parameter, but that should be pretty easy to add now - I just didn't get around to it. I'm on holiday at the moment, but I'll accept a pull request (you just need to add a parameter [here](https://github.com/benjamin-hodgson/asynqp/blob/master/src/asynqp/channel.py#L40) and pass the argument down to [here](https://github.com/benjamin-hodgson/asynqp/blob/master/src/asynqp/channel.py#L336)).

However I'm also confused as to what you're trying to achieve in the first place. Why do you want to publish messages to an exchange which might not exist?

**Comment by OscaTutenchamon** [comment]
I know I've used library not exactly as was it intended, it's just I didn't wanted to play around with many (for example) exchange objects but just execute some command knowing exchange's name alone. I think I have to rethink my design...

For the missing feature, it was a main reason I started to play around with undocumented, internal API. I might try to add it to `asynqp` :)

Thanks for quick reply! :+1:
